### PR TITLE
SSL for PGSQL driver

### PR DESCRIPTION
**Pull request by mkacper**
This PR enables PGSQL driver to establish SSL connections with a PostgreSQL database. What is more, the PR setups dockerized PostgreSQL on TravisCI and forces the database and the driver to use only SSL connections in tests. Documentation of the feature is also provided.

### CODE REVIEW

File: apps/ejabberd/src/mongoose_metrics.erl [hunk 1]
```diff
@@ -194,14 +194,23 @@ get_odbc_stats(ODBCWorkers) ->
     PortStats = [inet_stats(Port) || Port <- lists:flatten(Ports)],
     [{workers, length(ODBCConnections)} | merge_stats(PortStats)].
 
-get_port_from_odbc_connection({{ok, DB, Pid}, WorkerPid}) when DB =:= mysql; DB =:= pgsql ->
```
Comment (mkacper) [inline apps/ejabberd/src/mongoose_metrics.erl]: This part is improved in PR #1556.

File: tools/docker-setup-postgres.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#bin/bash
```
Comment (DenysGonchar) [inline tools/docker-setup-postgres.sh]: [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) starts with #!

File: apps/ejabberd/src/mongoose_metrics.erl [hunk 3]
```diff
@@ -194,14 +194,23 @@ get_odbc_stats(ODBCWorkers) ->
...
+get_port_from_odbc_connection({{ok, DB, Pid}, _WorkerPid}) when DB =:= pgsql ->
+    ProcStatus = sys:get_status(Pid),
+    get_port_from_proc_status(ProcStatus);
 get_port_from_odbc_connection({{ok, odbc, Pid}, WorkerPid}) ->
     Links = element(2, erlang:process_info(Pid, links)) -- [WorkerPid],
     [Port || Port <- Links, is_port(Port), {name, "tcp_inet"} == erlang:port_info(Port, name)];
 get_port_from_odbc_connection(_) ->
     undefined.
 
+get_port_from_proc_status({_, _, _, Status}) ->
```
Comment (DenysGonchar) [inline apps/ejabberd/src/mongoose_metrics.erl]: this function is awful, but i's fixed in another PR. so we can leave it as is here

File: doc/Advanced-configuration.md [hunk 4]
```diff
@@ -270,6 +270,34 @@ See [Database backends configuration](./advanced-configuration/database-backends
     * **Supported values:** `mssql`, `generic`
     * **Default:** `generic`
 
+#### PostgreSQL SSL connections
+
+In order to establish a secure connection with a PostgreSQL database additional options must be passed in aforementioned `odbc_server` tuple.
+Here is the proper syntax:
+
+`{odbc_server, {Type, Host, Port, DBName, Username, Password, SSL}`
```
Comment (erszcz) [inline doc/Advanced-configuration.md]: Missing closing `}`.

**Pull request merged**